src/libostree/ostree-repo.c

```diff
@@ -1266,6 +1266,86 @@ impl_repo_remote_merge (OstreeRepo     *self,
   return TRUE;
 }
 
+static gboolean
+impl_repo_remote_replace (OstreeRepo     *self,
+                          GFile          *sysroot,
+                          const char     *name,
+                          const char     *url,
+                          GVariant       *options,
+                          GCancellable   *cancellable,
+                          GError        **error)
+{
+  g_return_val_if_fail (name != NULL, FALSE);
+  g_return_val_if_fail (url != NULL, FALSE);
+  g_return_val_if_fail (options == NULL || g_variant_is_of_type (options, G_VARIANT_TYPE ("a{sv}")), FALSE);
+
+  if (!ostree_validate_remote_name (name, error))
+    return FALSE;
+
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(OstreeRemote) remote = _ostree_repo_get_remote (self, name, &local_error);
+  if (remote == NULL)
+    {
+      if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
+        }
```

Comment:
`g_clear_error (&local_error)` here.

Reply:
Originally I had that, but is it necessary here? It's declared with `g_autoptr` and never reused in this branch. But I can add it to be sure it doesn't leak.

Reply:
Yes, but at this point the error is set - so we must clear it for the invocation of `impl_repo_remote_add()` right below, otherwise get could get the overwriting-error error.

Reply:
The call to `impl_repo_remote_add` uses `error`, not `local_error`. `local_error` is only used to check if the repo exists. The caller's `error` is used when adding and `local_error` will be cleared regardless of what happens there. I don't mind clearing the error, but I don't think it's really necessary.

Reply:
Ah yes, you're totally right.  That said, I still think it's best practice to clear checked errors right after checking them.  Imagine if someone later comes and wants to add another checked call, sees there's already a `local_error` value and doesn't realize it may have been written earlier.